### PR TITLE
Feature/#38 Create user settings

### PR DESCRIPTION
**Pull request by StepanovRoma**
1. Made a fully working page with settings, validation and display
2. Reworked authorization and now after refreshing the page the user remains logged in (redux)
3. It needs a lot of testing
4. Perhaps refactoring is definitely needed, everything will be added to separate tasks, the auth duck also needs to be finalized
image

### CODE REVIEW

**Comment by StepanovRoma** [comment]
[Create user settings](https://trello.com/c/e3uOvVfL/38-create-user-settings)

**Review by MangriMen — APPROVED** [review]

**Review by FFreestyler — APPROVED** [review]

File: src/schemas/auth.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import * as yup from 'yup';
 
 export const registerSchema = yup.object({
-  username: yup.string().required('Username is required'),
+  username: yup
+    .string()
+    .max(20, 'nick should be less 20 chars')
```
Comment (IHiFiveI) [inline src/schemas/auth.ts:6]: Replace with "Nickname can't be longer than 20 characters"

File: src/locales/ru/translation.json [hunk 2]
```diff
@@ -23,10 +25,17 @@
   },
   "userProfile": {
     "name": "Имя",
-    "email": "Почта",
+    "email": "Email",
```
Comment (IHiFiveI) [inline src/locales/ru/translation.json]: Return old translation

File: src/ducks/userSettings/index.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      state.loading = {
+        username: false,
+        email: false,
+        password: false,
+        image: false,
+      };
+    },
+    updateUserFailed(state, action) {
+      state.error = action.payload;
+      state.loading = {
```
Comment (IHiFiveI) [inline src/ducks/userSettings/index.ts]: Replace with `state.loading = initialState.loading`

File: src/ducks/user/index.ts [hunk 4]
```diff
@@ -1,17 +1,39 @@
...
+      state.isLoading = false;
+      state.user.avatar = action.payload.avatar;
+      state.user.username = action.payload.username;
+      state.user.email = action.payload.email;
+    },
+    fetchUserFailed(state, action) {
+      state.isLoading = false;
+      state.user.avatar = '';
+      state.user.username = '';
+      state.user.email = '';
```
Comment (IHiFiveI) [inline src/ducks/user/index.ts]: Replace using initialState.user

File: src/ducks/auth/index.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { auth } from 'configs/firebase';
 
-const initialState = { isLogged: false, error: '' };
+const initialState = {
+  isLogged: !auth?.currentUser,
```
Comment (MangriMen) [inline src/ducks/auth/index.ts]: isLogged should be changed to false, becuase firebase is loading slower than initialState and it always be false.

File: src/ducks/auth/index.ts [hunk 6]
```diff
@@ -19,7 +23,7 @@ export const authSlice = createSlice({
     register: (state, action) => {
       state.error = '';
     },
-    registerSuccess: (state) => {
+    registerSuccess: (state, action) => {
```
Comment (MangriMen) [inline src/ducks/auth/index.ts]: Remove unnecessary action.

File: src/components/App.tsx [hunk 7]
```diff
@@ -1,14 +1,23 @@
...
 import NotFoundPage from './pages/NotFoundPage';
 import UserPage from './pages/UserPage';
+import { UserSettingsComponent } from './user/UserSettingsComponent';
 
 const App: FC = () => {
+  const dispatch = useDispatch();
+  auth?.onAuthStateChanged(() => {
+    dispatch(fetchUser());
+  });
+
```
Comment (MangriMen) [inline src/components/App.tsx]: Add user argument to arrow function and check it not to be null; add dispatch loginSuccess action. Move fetchUser and loginSuccess dispatches to if statement with user check (like `if (user) {}`) and add logoutSuccess dispacth to else branch of if statement.

**Review by MangriMen — CHANGES_REQUESTED** [review]

**Review by MangriMen — APPROVED** [review]

**Review by IHiFiveI — APPROVED** [review]

**Pull request merged**